File: YemenSchoolsV1.Domain/Entities/AcademicYear.cs
namespace YemenSchoolsV1.Domain.Entities
{
    public class AcademicYear
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public bool IsCurrentYear { get; set; } = false;

        public Guid SchoolId { get; set; }

        // Navigation Property
        public School School { get; set; } = null!;
        public ICollection<Term> Terms { get; set; } = [];
        public ICollection<Section> Sections { get; set; } = [];
        public ICollection<Student> Students { get; set; } = [];
        public ICollection<Attendance> Attendances { get; set; } = [];



    }
}

File: YemenSchoolsV1.Clint/src/app/features/school-dashboard/year/models/AcademicYear.ts
export interface YearDto {
    id: string;
    name: string;
    startDate: Date;
    endDate: Date;
    isCurrentYear: boolean
    isCurrentYearDisplay?: string;
}

export interface CreateYearDto {
    id: string;
    name: string;
    schoolId: string;
    startDate: Date;
    endDate: Date;
}
